File: android/app/src/main/java/com/example/android/viewmodel/home/HomeViewModel.kt
package com.example.android.viewmodel.home

import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.setValue
import androidx.lifecycle.ViewModel

data class HomeMetric(
    val label: String,
    val value: String,
    val sublabel: String
)

data class RouteStop(
    val index: Int,
    val name: String,
    val address: String,
    val distance: String,
    val timeEstimate: String,
    val itemsLabel: String
)

data class HomeUiState(
    val userName: String = "John Doe",
    val savingsThisTrip: String = "$6.70",
    val totalBudget: String = "$120.00",
    val budgetUsed: String = "$57.31",
    val savedThisMonth: String = "$42.80",
    val savedThisMonthLabel: String = "this month",
    val avgTripTime: String = "34m",
    val milesSaved: String = "12.4",
    val itemsTracked: String = "89",
    val alertsCount: String = "3",
    val optimizedStopsLabel: String = "3 stops",
    val metrics: List<HomeMetric> = emptyList(),
    val routeStops: List<RouteStop> = emptyList()
)

class HomeViewModel : ViewModel() {
    var uiState by mutableStateOf(
        HomeUiState(
            metrics = listOf(
                HomeMetric("Budget", "$57.31", "of $120.00"),
                HomeMetric("Saved", "$42.80", "this month"),
                HomeMetric("Avg trip", "34m", ""),
                HomeMetric("Miles saved", "12.4", "miles saved"),
                HomeMetric("Items", "89", "tracked"),
                HomeMetric("Alerts", "3", "alerts")
            ),
            routeStops = listOf(
                RouteStop(1, "Aldi", "142 Atlantic Ave", "0.8 mi", "12 min", "3 items"),
                RouteStop(2, "Trader Joe's", "130 Court St", "1.2 mi", "8 min", "4 items"),
                RouteStop(3, "Costco", "976 3rd Ave", "2.4 mi", "15 min", "2 items")
            )
        )
    )
        private set
}

File: ios/Neighborly/Neighborly/ViewModels/HomeViewModel.swift
import Foundation
import SwiftUI

struct HomeMetric: Identifiable {
    let id = UUID()
    let label: String
    let value: String
    let sublabel: String
    let icon: String
    let tint: MetricTint

    enum MetricTint {
        case green, orange, gray
    }
}

struct HomeRouteStop: Identifiable {
    let id = UUID()
    let index: Int
    let name: String
    let address: String
    let distance: String
    let timeEstimate: String
    let itemsLabel: String
}

@Observable
final class HomeViewModel {
    let userName: String = "John Doe"
    let savingsThisTrip: String = "6.70"
    let totalBudget: String = "$120.00"
    let budgetUsed: String = "$57.31"
    let budgetProgress: Double = 57.31 / 120.0
    let savedThisMonth: String = "$42.80"
    let savedThisMonthLabel: String = "this month"
    let avgTripTime: String = "34m"
    let milesSaved: String = "12.4"
    let itemsTracked: String = "89"
    let alertsCount: String = "3"
    let optimizedStopsLabel: String = "3 stops"

    let savingsBarHeights: [CGFloat] = [0.4, 0.7, 0.5, 0.9, 0.6]

    let routeStops: [HomeRouteStop] = [
        HomeRouteStop(index: 1, name: "Aldi", address: "142 Atlantic Ave", distance: "0.8 mi", timeEstimate: "12 min", itemsLabel: "3 items"),
        HomeRouteStop(index: 2, name: "Trader Joe's", address: "130 Court St", distance: "1.2 mi", timeEstimate: "8 min", itemsLabel: "4 items"),
        HomeRouteStop(index: 3, name: "Costco", address: "976 3rd Ave", distance: "2.4 mi", timeEstimate: "15 min", itemsLabel: "2 items")
    ]
}
